### PR TITLE
apis/nfd/validate: loosen validation of feature annotations

### PR DESCRIPTION
**Pull request by marquiz**
Don't require that the annotation value must conform to the (strict)
requirements of label values. In the Kubernetes API annotation values do
not have other restrictions than that the total size (keys and values)
of _all_ annotations combined of an object must not exceed 256kB.

This patch sets a maximum size limit of 1kB for the value of a single
feature annotation created by NFD. This limit is rather arbitrary but
should be enough for the NFD usage scenarios (until proven wrong).

Also add license header to the test.go file and fix one bug in
MatchFeature validation.

### CODE REVIEW

File: pkg/apis/nfd/validate/validate_test.go [hunk 1]
```diff
@@ -1,10 +1,27 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (ArangoGutierrez) [inline pkg/apis/nfd/validate/validate_test.go:2]: 24?

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Just a small question

File: pkg/apis/nfd/validate/validate_test.go (hunk 1 above)
Comment (marquiz) [reply pkg/apis/nfd/validate/validate_test.go:2]: The file was created in 2023 (by you, I think 😅) but the license header was missing. So I put the year of the original contribution here. Makes sense?

**Comment by marquiz** [comment]
Rebased. @ArangoGutierrez PTAL

**Comment by marquiz** [comment]
ping @ArangoGutierrez

File: pkg/apis/nfd/validate/validate.go [hunk 2]
```diff
@@ -79,6 +80,13 @@ func Template(labelsTemplate string) []error {
 	return validationErr
 }
 
+func sortErrors(errs []error) []error {
```
Comment (PiotrProkop) [inline pkg/apis/nfd/validate/validate.go:83]: shouldn't we just invoke this func in tests? what's the value in sorting errors by char len?

Comment (marquiz) [reply pkg/apis/nfd/validate/validate.go]: It doesn't sort by length but alphabetically.

Sorting can be done in tests, yes. I moved it there.

**Comment by marquiz** [comment]
PR updated. @PiotrProkop PTAL

**Comment by marquiz** [comment]
Thanks @PiotrProkop for the review!

**Pull request merged**